grammar.js

```diff
@@ -1,337 +1,411 @@
-    'mul',
-    'div',
-    'rem',
-    'lt',
-    'leq',
-    'gt',
-    'geq',
-    'eq',
-    'neq',
-    'pad',
-    'asUInt',
-    'asAsyncReset',
-    'asSInt',
-    'asClock',
-    'shl',
-    'shr',
-    'dshl',
-    'dshlw',
-    'dshr',
-    'dshrw',
-    'cvt',
-    'neg',
-    'not',
-    'and',
-    'or',
-    'xor',
-    'andr',
-    'orr',
-    'xorr',
-    'cat',
-    'bits',
-    'head',
-    'tail',
-    'asFixedPoint',
-    'bpshl',
-    'bpshr',
-    'bpset'
-  ),
-
-  // Tokens
-
-  UnsignedInt: $ => choice('0', /[1-9][0-9]*/),
-
-  SignedInt: $ => seq(choice('+', '-'), /[1-9][0-9]*/),
-
-  // PosInt: $ => /[1-9][0-9]*/,
-
-  HexLit: $ => /"h[+-]?[a-fA-F0-9]+"/,
-
-  OctLit: $ => /"o[+-]?[0-7]+"/,
-
-  BinLit: $ => /"b[+-]?[01]+"/,
-
-  DoubleLit: $ => /[+-]*[0-9]+\.[0-9]+(E[+-]?[0-9]+)?/,
-
-  StringLit: $ => /"([\x09\x20\x21\x23-\xFE]|\\")*"/,
-
-  RawString: $ => /'[^\\']*'/,
-
-  FileInfo: $ => /@\[.*\]/,
-
-  // Id: $ => /[a-zA-Z_][a-zA-Z0-9$]*/, // LegalStartChar (LegalIdChar)*
-  id: $ => /[a-zA-Z_][a-zA-Z0-9_$]*/, // LegalStartChar (LegalIdChar)*
-
-  RelaxedId: $ => /[a-zA-Z0-9_$]+/, // (LegalIdChar)+
-
-  comment: $ => token(
-    seq(';', /.*/)
-  )
-
-};
+    ),
 
-module.exports = grammar({
-  name: 'firrtl',
-  word: $ => $.id,
-  rules: rules,
-  extras: $ => [/\s/, $.comment],
-  externals: $ => [
-    $._newline,
-    $._indent,
-    $._dedent
-  ]
-});
+    stop: $ => seq('stop', '(', $.expression, ',', $.expression, ',', $.number, ')', optional($.info)),
+
+    printf: $ => seq(
+      'printf',
+      '(',
+      $.expression, ',',
+      $.expression, ',',
+      $.string,
+      repeat(seq(',', $.expression)),
+      ')',
+      optional($.info)
+    ),
+
+    verif: $ => seq(
+      choice('assert', 'assume', 'cover'),
+      '(',
+      $.expression, ',',
+      $.expression, ',',
+      $.expression, ',',
+      $.string,
+      ')'
+    ),
+
+    skip: $ => seq('skip', optional($.info)),
+
+    attach: $ => seq('attach', '(', sep1(',', $.expression), ')', optional($.info)),
+
+    info: _ => /@\[.*\]/, // $.FileInfo,
+
+    mdir: _ => choice('infer', 'read', 'write', 'rdwr'),
+
+    ruw: _ => choice('old', 'new', 'undefined'),
+
+    litType: $ => seq(
+      choice('UInt', 'SInt'),
+      optional(seq('<', $.number, '>'))
+    ),
+
+    expression: $ => choice(
+      $.literal,
+      $.identifier,
+      $.keyword_identifier,
+      $.sub_field,
+      $.sub_index,
+      $.sub_access,
+      $.mux,
+      $.conditionally_valid,
+      $.primitive_operation
+    ),
+
+    literal: $ => seq($.litType, '(', choice($.number, $.number_str), ')'),
+
+    sub_field: $ => seq($.expression, '.', $.field_id),
+
+    sub_index: $ => seq($.expression, '[', $.number, ']'),
+
+    sub_access: $ => seq($.expression, '[', $.expression, ']'),
+
+    mux: $ => seq('mux', '(', $.expression, ',', $.expression, ',', $.expression, ')'), // no commas in ANTLR
 
-/* eslint camelcase: 0 */
-/* eslint no-undef: 0 */
```

Comment:
`no-undef` and `no-unused-vars` was in this file because they are specific to this file only.